lib/opts/index.ts

```diff
@@ -0,0 +1,122 @@
+import { getConfiguration as getConfigFile } from "../configuration";
+import { isin } from "../../type-helpers";
+import type { PrettyPrint } from "../../type-helpers";
+
+type Release = "minor" | "major" | "patch";
+type Task = "changelog" | "commit" | "tag";
+type Hook =
+  | "prerelease"
+  | "prebump"
+  | "postbump"
+  | "prechangelog"
+  | "postchangelog"
+  | "precommit"
+  | "postcommit"
+  | "pretag"
+  | "posttag";
+
+type TypePrefixes = Array<
+  (
+    | { section: string; hidden?: boolean | undefined }
+    | { hidden: true; section?: string | undefined }
+  ) & { type: string }
+>;
+
+type ConfigFiles = Array<
+  | string
+  | { filename: string; type: "json" | "gradle" | "plain-text" }
+  | { filename: string; updater: string }
+>;
+
+/**
+ * __THIS SHOULD NOT CHANGE.__
+ * @deprecated
+ *
+ * The configuration options for `standard-version` as of version 9.5 (The last version prior to the fork; deprecated).
+ */
+export type SVConfig = {
+  packageFiles: ConfigFiles;
+  bumpFiles: ConfigFiles;
+  releaseAs: Release;
+  prerelease: string | boolean;
+  infile: string;
+  message: string;
+  firstRelease: boolean;
+  sign: boolean;
+  noVerify: boolean;
+  commitAll: boolean;
+  silent: boolean;
+  tagPrefix: string;
+  scripts: Record<Hook, string>;
+  skip: Record<Task, string>;
+  dryRun: boolean;
+  gitTagFallback: boolean;
+  path: string;
+  changelogHeader: string;
+  preset: string;
+  lernaPackage: string;
+  header: string;
+  types: TypePrefixes;
+  preMajor: boolean;
+  commitUrlFormat: string;
+  compareUrlFormat: string;
+  issueUrlFormat: string;
+  userUrlFormat: string;
+  releaseCommitMessageFormat: string;
+  issuePrefixes: string[];
+};
+
+/**
+ * The configuration object for commit-and-tag-version, which is a superset of the conventional-changelog-config-spec (as of version 2.1.0)
+ * This may or may not maintain backwards compatibility with standard-version (as of version 9.5.0).
+ */
+export type CatVConfig = PrettyPrint<
+  SVConfig & {
+    npmPublishHint: string;
+    releaseCount: number;
+    tagForce: boolean;
+  }
+>;
+
+/** The configuration options that are not supported by standard-version (as of version 9.5.0). */
+const catVOnlyFeatures = [
+  "npmPublishHint",
+  "releaseCount",
+  "tagForce",
+] as const satisfies ReadonlyArray<Exclude<keyof CatVConfig, keyof SVConfig>>;
+
+export const getMergedConfig = async (
+  cwd?: string
+): Promise<Partial<CatVConfig>> => {
+  const dir = cwd ?? process.cwd();
+  const json = (await import("path")).join(dir, "package.json");
+  const svConfig: SVConfig = (await import(json))["standard-version"];
+  const catvConfig: CatVConfig = (await import(json))["commit-and-tag-version"];
+
+  const config: Partial<CatVConfig> = {};
+
+  if (svConfig) {
+    for (const key in svConfig) {
+      // @ts-expect-error - We know this key SHOULD NOT be in svConfig, but we're going to allow it and warn the user
```

Comment:
You can probably avoid all the `// @ts-expect-error` directives by making this an `Object.keys(svConfig).filter(() => /* etc */)`

Reply:
Unfortunately, I don't believe `.filter()`acts as a type guard. There is a library, `ts-reset`, that does this, but it's not recommended for libraries because it pollutes the global scope.

Reply:
I agree, let's not pollute the global scope 😅 

What I mean is:

```
Object.keys(svConfig)
  .filter((key) => catVOnlyFeatures.includes(key))
  .forEach((key) => console.warn(
          `The "${key}" option is a feature of commit-and-tag-version, and is not supported by standard-version.${"\n"}Please move this option to the 'commit-and-tag-version' key.${"\n"}In a future version, this will throw an error.`))
```

Reply:
I understand your meaning, but what I'm saying is that - despite how one might *expect* .filter() and Object.keys() to work, the type of `key` remains as `string` throughout that pipeline. It is never narrowed to the union of literal strings that are keys of svConfig, which is what is needed to remove the type errors.

You aren't the only one to expect this behavior, though. I certainly did, as did many others. That's why I bring up `ts-reset` - it was written with the purpose of making some less intuitive behaviors act how you would expect, and I think this is one of the things it addresses (again, not advocating for it here lol).

Reply:
Wow - ignore me. I feel dumb now! I think I understand what you mean. I've refactored like so:

```
export const getMergedConfig = async (
  cwd?: string
): Promise<Partial<Config>> => {
  const searchDir = cwd ?? process.cwd();
  const pkgJson = (await import("path")).join(searchDir, "package.json");
  const legacyConf: LegacyConfig = (await import(pkgJson))["standard-version"];
  const modernConf: Config = (await import(pkgJson))["commit-and-tag-version"];

  Object.keys(legacyConf ?? {}).forEach((key) => {
    if (catVOnlyFeatures.includes(key as any)) {
      console.warn(
        `The "${key}" option is a feature of commit-and-tag-version, and is not supported by standard-version.${"\n"}Please move this option to the 'commit-and-tag-version' key.${"\n"}In a future version, this will throw an error.`
      );
    }
    if (modernConf && isin(modernConf, key as any)) {
      console.warn(
        `"standard-version"."${key}" in package.json is being overridden by "commit-and-tag-version"."${key}". in package.json`
      );
    }
  });

  const configFromFile = await getConfigFromFile();
  return { ...(legacyConf ?? {}), ...(modernConf ?? {}), ...(configFromFile ?? {}) };
};
```

That about right?